src/poetry/core/semver/version.py

```diff
@@ -92,8 +92,6 @@ def allows(self, version: Version | None) -> bool:
         # allow weak equality to allow `3.0.0-1` for `3.0.0`
         if not _this.is_postrelease() and _other.is_postrelease():
             _other = _other.without_postrelease()
-        elif _this.without_postrelease() and not _other.without_postrelease():
```

Comment:
Think this should be

```py
elif _this.is_postrelease() and not _other.is_postrelease():
```

Reply:
This is what I originally assumed, but this made the following 2 tests fail in [test_allows_with_post](https://github.com/mkniewallner/poetry-core/blob/f26e2361f0d643826d31366887dcb425fc078214/tests/semver/test_version.py#L134-L141):
```python
assert not v.allows(Version.parse("1.2.3"))
assert not v.allows(Version.parse("1.2.3+build.2"))
```
Which led me to believe that the current tests are correct, and that this code branch may well be unnecessary.

Reply:
I wonder if that is exposing https://github.com/python-poetry/poetry/issues/4729#issuecomment-1110930059

Reply:
But that said `1.2.3-1` shouldn't in fact allow `1.2.3` or `1.2.3+local`, so you might be correct.